samples/core/tutorials/estimators/wide.ipynb

```diff
@@ -0,0 +1,1337 @@
+      ]
+    },
+    {
+      "metadata": {
+        "id": "tTwQzWcn8aBu",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "! git clone --depth 1 https://github.com/tensorflow/models"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "yVvFyhnkcYvL",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "models_path = os.path.join(os.getcwd(), 'models')\n",
+        "\n",
+        "sys.path.append(models_path)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "15Ethw-wcYvP",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Execute the data download script:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "6QilS4-0cYvQ",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "from official.wide_deep import census_dataset\n",
+        "from official.wide_deep import census_main\n",
+        "\n",
+        "census_dataset.download(\"/tmp/census_data/\")"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "cD5e3ibAcYvS",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "To execute the tutorial code from the command line first add the path to tensorflow/models to your `PYTHONPATH`."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "DYOkY8boUptJ",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "#export PYTHONPATH=${PYTHONPATH}:\"$(pwd)/models\"\n",
+        "os.environ['PYTHONPATH'] += os.pathsep+models_path"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "5r0V9YUMUyoh",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Use `--help` to see what command line options are available: "
+      ]
+    },
+    {
+      "metadata": {
+        "id": "1_3tBaLW4YM4",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "!python -m official.wide_deep.census_main --help"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "RrMLazEN6DMj",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Now run the model:\n"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "py7MarZl5Yh6",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "!python -m official.wide_deep.census_main --model_type=wide --train_epochs=2"
```

Comment:
@karmel 

In colab this just hangs for about a minute before running.
It doesn't do that on my local machine.

Any ideas?

If I interrupt it the traceback is (`cloud_lib.on_gcp()`):

```
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/content/models/official/wide_deep/census_main.py", line 112, in <module>
    absl_app.run(main)
  File "/usr/local/lib/python3.6/dist-packages/absl/app.py", line 274, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.6/dist-packages/absl/app.py", line 238, in _run_main
    sys.exit(main(argv))
  File "/content/models/official/wide_deep/census_main.py", line 106, in main
    run_census(flags.FLAGS)
  File "/content/models/official/wide_deep/census_main.py", line 101, in run_census
    early_stop=True)
  File "/content/models/official/wide_deep/wide_deep_run_loop.py", line 91, in run_loop
    test_id=flags_obj.benchmark_test_id)
  File "/content/models/official/utils/logs/logger.py", line 151, in log_run_info
    test_id))
  File "/content/models/official/utils/logs/logger.py", line 317, in _gather_run_info
    _collect_test_environment(run_info)
  File "/content/models/official/utils/logs/logger.py", line 421, in _collect_test_environment
    if cloud_lib.on_gcp():
  File "/content/models/official/utils/logs/cloud_lib.py", line 28, in on_gcp
    response = requests.get(GCP_METADATA_URL, headers=GCP_METADATA_HEADER)
  File "/usr/local/lib/python3.6/dist-packages/requests/api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.6/dist-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/requests/adapters.py", line 440, in send
    timeout=timeout
  File "/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py", line 357, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/lib/python3.6/http/client.py", line 1239, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1285, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1234, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.6/http/client.py", line 1026, in _send_output
    self.send(msg)
  File "/usr/lib/python3.6/http/client.py", line 964, in send
    self.connect()
  File "/usr/local/lib/python3.6/dist-packages/urllib3/connection.py", line 166, in connect
    conn = self._new_conn()
  File "/usr/local/lib/python3.6/dist-packages/urllib3/connection.py", line 141, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py", line 73, in create_connection
    sock.connect(sa)
```

Reply:
Hmm, I'm not that familiar with colab-- could it be that the test is timing out? Or not finding the data or something?

Reply:
It has the data. It runs fine. It just waits at this line for a minute first like it's waiting for this connection to time out.

This is in `benchmark_logger.log_run_info` --> what is that? down inside `cloud_lib.on_gcp()`.
(It's not doing it in the rest of the notebook because I don't use benchmarks)

I'll send this to the Colab people and see what they say.

Reply:
From the stack trace. It seems that the logger was trying to detect the local running env, by querying URL "http://metadata/computeMetadata/v1/instance/hostname". Maybe the URL is not reachable on the codelab box, and suffering some delay when querying.

I probably should add a short timeout for the connection.

Reply:
https://github.com/tensorflow/models/pull/4762

Reply:
Oh, nice. Thanks!